model/markers.go

```diff
@@ -0,0 +1,286 @@
+package model
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/wasabee-project/Wasabee-Server/log"
+	"github.com/wasabee-project/Wasabee-Server/messaging"
+)
+
+// MarkerID wrapper to ensure type safety
+type MarkerID string
+
+// MarkerType will be an enum once we figure out the full list
+type MarkerType string
+
+// Marker is defined by the Wasabee IITC plugin.
+type Marker struct {
+	ID         MarkerID   `json:"ID"`
+	PortalID   PortalID   `json:"portalId"`
+	Type       MarkerType `json:"type"`
+	AssignedTo GoogleID   `json:"assignedTo"` // deprecated, use Assignments from Task
+	Task
+}
+
+// TODO use the logic from insertZone to unify insertMarker and updateMarker
+
+// insertMarkers adds a marker to the database
+func (opID OperationID) insertMarker(m Marker) error {
+	if m.State == "" {
+		m.State = "pending"
+	}
+
+	if !m.Zone.Valid() || m.Zone == ZoneAll {
+		m.Zone = zonePrimary
+	}
+
+	_, err := db.Exec("REPLACE INTO task (ID, opID, comment, taskorder, state, zone, delta) VALUES (?, ?, ?, ?, ?, ?, ?)",
+		m.ID, opID, MakeNullString(m.Comment), m.Order, m.State, m.Zone, m.DeltaMinutes)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	_, err = db.Exec("REPLACE INTO marker (ID, opID, PortalID, type) VALUES (?, ?, ?, ?)", m.ID, opID, m.PortalID, m.Type)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	// until clients get updated
+	if m.AssignedTo != "" {
+		m.Assignments = append(m.Assignments, m.AssignedTo)
+	}
+
+	if len(m.Assignments) > 0 {
+		err = m.Assign(m.Assignments, nil)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+	}
+
+	if len(m.DependsOn) > 0 {
```

Comment:
do we need to distinguish empty array (all depends removed) from missing depends (old client) ?
(same for assignments)

Reply:
Due to how Go's json library works, that would be quit a lot of work.

I don't think it will be too much of an issue in reality. Those who want to use this feature will be the first to upgrade.